projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -0,0 +1,201 @@
+#include "ads1015.h"
+#include <status.h>
+#include <stdio.h>
+#include "ads1015_def.h"
+#include "gpio_it.h"
+
+static void prv_check_status(StatusCode code) {
+  if (!code) {
+    status_msg(code, "failed");
+  }
+}
+
+// reset the internal registers and enter a power-down state
+static StatusCode prv_reset(ADS1015Data *data) {
+  uint8_t reset = ADS1015_RESET_BYTE;
+  StatusCode code = i2c_write(data->i2c_port, ADS1015_I2C_GENERAL_CALL, &reset, sizeof(reset));
+  prv_check_status(code);
```

Comment:
`prv_check_status` would do nothing - it cannot return from the function.

The macro below (`status_ok_or_return`) is actually what works. However, in this case, you might as well just return the StatusCode from `i2c_write` directly. If you expand the macro, it looks something like

```
if (code != STATUS_CODE_OK) {
  return code;
}
return STATUS_CODE_OK;
```

which doesn't really make any sense.